cli/azd/pkg/auth/manager.go

```diff
@@ -361,10 +374,14 @@ func (m *Manager) GetLoggedInServicePrincipalTenantID(ctx context.Context) (*str
 func (m *Manager) newCredentialFromClientSecret(
 	tenantID string,
 	clientID string,
-	clientSecret string) (azcore.TokenCredential, error) {
+	clientSecret string,
+) (azcore.TokenCredential, error) {
 	options := &azidentity.ClientSecretCredentialOptions{
 		ClientOptions: policy.ClientOptions{
 			Transport: m.httpClient,
+			// TODO: Inject client options instead? this can be done if we're OK
+			// using the default user agent string.
+			Cloud: m.cloud.Configuration,
```

Comment:
@weikanglim should we use the options object here or the one defined in line 114? The object-level options does involve a cache.

Reply:
This can be handled as part of making sure the user agent is consistent: https://github.com/Azure/azure-dev/issues/2723